### PR TITLE
[Feature] Support basic submit syntax

### LINKED ISSUE #5845: Support basic submit syntax
support
```
SUBMIT [/*+ SET_VAR(key=value) */] TASK [[database.]<task_name>] 
AS
<CTAS>;
```

### PR DESCRIPTION
**Pull request by Astralidea**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #5845

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 0 / 60 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|0|9|00.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|0|18|00.00%|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/ast/SubmitTaskStmt.java|0|22|00.00%|
|:large_blue_circle:|com/starrocks/qe/DdlExecutor.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/TaskAnalyzer.java|0|8|00.00%|

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 [hunk 1]
```diff
@@ -109,6 +112,14 @@ dropViewStatement
     : DROP VIEW (IF EXISTS)? qualifiedName
     ;
 
+// ------------------------------------------- Task Statement ----------------------------------------------------------
+
+submitTaskStatement
+    : SUBMIT TASK qualifiedName?
+    sessionPropertis?
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:119]: Do we need to support session in this version?

Comment (imay) [reply fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:119]: I think user can change session variable by /* SET_VAR= */. we don't need to give them more options

Comment (Astralidea) [reply fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:119]: I agree with you

**Review by gengjun-git — DISMISSED** [review]

**Comment by Astralidea** [comment]
run all

**Comment by Astralidea** [comment]
run all

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 51 / 64 (79.69%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/qe/DdlExecutor.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/sql/ast/SubmitTaskStmt.java|15|22|68.18%|
|:large_blue_circle:|com/starrocks/sql/analyzer/TaskAnalyzer.java|6|8|75.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|21|22|95.45%|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|9|9|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 [hunk 2]
```diff
@@ -572,6 +582,10 @@ distributionDesc
     : DISTRIBUTED BY HASH identifierList (BUCKETS INTEGER_VALUE)?
     ;
 
+sessionPropertis
```
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:585]: unused?

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocksLex.g4 [hunk 3]
```diff
@@ -179,6 +180,7 @@ SYSTEM: 'SYSTEM';
 TABLE: 'TABLE';
 TABLES: 'TABLES';
 TABLET: 'TABLET';
+TASK: 'TASK';
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocksLex.g4:183]: should be a unReserved word.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 51 / 64 (79.69%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/qe/DdlExecutor.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/sql/ast/SubmitTaskStmt.java|15|22|68.18%|
|:large_blue_circle:|com/starrocks/sql/analyzer/TaskAnalyzer.java|6|8|75.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|21|22|95.45%|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|9|9|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocksLex.g4 [hunk 4]
```diff
@@ -161,6 +161,7 @@ ROLLUP: 'ROLLUP';
 ROW: 'ROW';
 ROWS: 'ROWS';
 ROW_NUMBER: 'ROW_NUMBER';
+SUBMIT: 'SUBMIT';
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocksLex.g4:164]: should be a unReserved too to keep compatible if there is a column named `submit`

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 51 / 64 (79.69%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/qe/DdlExecutor.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/sql/ast/SubmitTaskStmt.java|15|22|68.18%|
|:large_blue_circle:|com/starrocks/sql/analyzer/TaskAnalyzer.java|6|8|75.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|21|22|95.45%|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|9|9|100.00%|

**Review by HangyuanLiu — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Pull request merged**